src/lib.rs

```diff
@@ -1274,6 +1290,29 @@ impl KeyPair {
 		let private_key_der :&[_] = &private_key.contents;
 		Ok(private_key_der.try_into()?)
 	}
+
+	/// Parses the key pair from a raw public key and a remote private key
+	pub fn from_remote(key_pair :Box<dyn RemoteKeyPair>) -> Result<Self, RcgenError> {
+		Ok(Self {
+			alg :key_pair.algorithm(),
+			kind :KeyPairKind::Remote(key_pair),
+			serialized_der : Vec::new(),
+		})
+	}
+}
+
+/// A private key that is not directly accessible, but can be used to sign messages
+///
+/// This is used to enable generating certificates from a remote and raw private key.
+pub trait RemoteKeyPair {
+	/// Return's the public key of this key pair
+	fn public_key(&self) -> &[u8];
+
+	/// Signs `msg` using the selected algorithm
+	fn sign(&self, msg :&[u8]) -> Result<Vec<u8>, RcgenError>;
+
+	/// Reveals which algorithm will be used when you call `sign()`
+	fn algorithm(&self) -> &'static SignatureAlgorithm;
```

Comment:
This is already stored in `KeyPair` anyway, I'm not sure if it's useful to have it in the trait at all.